controller/lib/hal/hal.h

```diff
@@ -207,7 +214,16 @@ class HalApi {
   void test_serialPutIncomingData(const char *data, uint16_t len);
 #endif
 
+  // Performs the device soft-reset
+  [[noreturn]] void reset_device();
+
+  // Pets the watchdog, this makes the watchdog not reset the
+  // system for configured amount of time
+  void watchdog_handler();
```

Comment:
If we're already deviating from the (IMO not great) Arduino names, I would suggest to make this name more descriptive - e.g. `renew_watchdog`